test/tests/functional/pbs_qrun.py

```diff
@@ -95,52 +95,51 @@ def test_invalid_host_val(self):
 
     def test_qrun_hangs(self):
         """
-        This test verifies whether qrun hangs, if scheduling is
-        disabled
+        This test submit 500 jobs with differnt equivalence class,
+        turn of scheduling and qrun job to
+        verify whether qrun hangs.
         """
         node = self.mom.shortname
-        time1 = int(time.time())
-        walltime = 1
+        err_msg = 'qrun: Not enough free nodes available'
+        joblist = []
         self.server.manager(MGR_CMD_SET, SCHED,
                             {'scheduling': 'False'})
-        err_msg = 'qrun: Not enough free nodes available'
-        while True:
-            self.server.manager(MGR_CMD_SET, NODE,
-                                {'resources_available.ncpus': 1}, id=node)
-
-            j1 = Job(TEST_USER)
-            j1.create_script('/bin/date')
-            a = {'Resource_List.walltime': walltime}
-            j1.set_attributes(a)
-            self.server.submit(j1)
-
-            walltime = walltime + 1
+        self.server.manager(MGR_CMD_SET, NODE,
+                            {'resources_available.ncpus': 1}, id=node)
+        for walltime in range(1, 501):
+            j = Job(TEST_USER)
+            j.create_script('/bin/date')
             a = {'Resource_List.walltime': walltime}
-            j1.set_attributes(a)
-            jid2 = self.server.submit(j1)
-            walltime = walltime + 1
-            time2 = int(time.time())
+            j.set_attributes(a)
+            jid = self.server.submit(j)
+            joblist.append(jid)
+        self.logger.info("Submitted 500 jobs with different walltime")
 
-            self.server.manager(MGR_CMD_SET, SCHED,
-                                {'scheduling': 'True'})
-            self.server.manager(MGR_CMD_SET, SCHED,
-                                {'scheduling': 'False'})
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 'True'})
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'scheduling': 'False'})
+        job = 10  # when we turn on scheduling, it scheduled few job.
+        while job < 501:
```

Comment:
I don't see job changing.  Won't this always be true?

Reply:
Yes as now the scheduling cycle is longer, so we dont need to turn scheduling off. I will address this in my next commit.

Reply:
No, we need to turn off scheduling. Because if its enabled, then qrun can trigger another cycle for the jobid we submitted to this.
The bug was, if the scheduling was disabled and qrun is submitted job in current sched cycle then it was hung. As it wasn't able to trigger another cycle, because before triggering we were checking whether scheduling is enabled. 
But as per the feature, irrespective of scheduling status, qrun should trigger the cycle.